modules/couchbase/couchbase.go

```diff
@@ -581,19 +654,35 @@ func (c *CouchbaseContainer) checkAllServicesEnabled(rawConfig []byte) bool {
 	return true
 }
 
-func exposePorts(enabledServices []service) []string {
-	exposedPorts := []string{MGMT_PORT + "/tcp", MGMT_SSL_PORT + "/tcp"}
+type serviceCustomizer struct {
+	enabledService Service
+}
 
-	for _, service := range enabledServices {
-		for _, port := range service.ports {
-			exposedPorts = append(exposedPorts, port+"/tcp")
-		}
+func (c serviceCustomizer) Customize(req *testcontainers.GenericContainerRequest) {
+	for _, port := range c.enabledService.ports {
+		req.ExposedPorts = append(req.ExposedPorts, port+"/tcp")
+	}
+}
+
+// withService creates a serviceCustomizer for the given service.
+// It's private to prevent users from creating other services than the Analytics and Eventing services.
+func withService(service Service) serviceCustomizer {
+	return serviceCustomizer{
+		enabledService: service,
 	}
+}
```

Comment:
@fbiville this is how I'm creating a custom implementation of a customizer for services, holding a state for the services to enable, and adding a Customize request method for the customizer which modifies the container request.

The services, because they are not part of the request, need to be transferred to the config above.